streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -107,8 +107,8 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
    */
   def onStop()
 
-  /** Override this to specify a preferred location (hostname). */
-  def preferredLocation : Option[String] = None
```

Comment:
Adding a new private var in receiver feels like the simplest workaround for the compatibility issue